cartoframes/auth/__init__.py

```diff
@@ -122,6 +121,10 @@ def set_default_credentials(
         _default_credentials.session = session
 
 
+def _is_url(text):
+    return re.match(r'^https?://.*$', text)
```

Comment:
What happens if the user enters `http://username.carto.com`? This will return False, but is that still a valid URL for API calls? If so, should we coerce it to `https` or give a warning?

Reply:
`http` should work, because of the `?`

Reply:
Works, I checked it the first time I saw it

Reply:
Great. Should we coerce to `https`?

Reply:
I am not sure. `base_url` is used mainly by onpremises, right? In this case, I suppose we shouldn't require `https`. What do you see it?